src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java

```diff
@@ -125,11 +139,50 @@ public CreateContainerCmdImpl(CreateContainerCmd.Exec exec, String image) {
      */
     @Override
     public CreateContainerResponse exec() throws NotFoundException, ConflictException {
+        //code flow taken from https://github.com/docker/docker/blob/master/runconfig/opts/parse.go
```

Comment:
I doubt that this logic should be part of the command. Commands in general are just dumb wrappers of the appropriate remote API resource (here: https://docs.docker.com/engine/reference/api/docker_remote_api_v1.22/#create-a-container). They should only take arguments/parameters that are needed to execute the request and nothing else.